tests/model/cycle_data/test_treeherder_cycling.py

```diff
@@ -0,0 +1,186 @@
+
+    call_command('cycle_data', 'from:treeherder', sleep_time=0, days=1, debug=True, chunk_size=1)
+
+    assert Job.objects.count() == 1
+    assert JobLog.objects.count() == (num_job_logs_before - num_job_logs_to_be_deleted)
+
+    for (object_type, objects) in extra_objects.values():
+        actual = set(item.id for item in object_type.objects.all())
+        expected = set(item.id for item in objects)
+        assert actual == expected
+
+
+def test_cycle_all_data_in_chunks(
+    test_repository, failure_classifications, sample_data, sample_push, mock_log_parser
+):
+    """
+    Test cycling the sample data in chunks.
+    """
+    job_data = sample_data.job_data[:20]
+    test_utils.do_job_ingestion(test_repository, job_data, sample_push, False)
+
+    # build a date that will cause the data to be cycled
+    cycle_date_ts = datetime.now() - timedelta(weeks=1)
+    for job in Job.objects.all():
+        job.submit_time = cycle_date_ts
+        job.save()
+
+    create_failure_lines(Job.objects.get(id=1), [(test_line, {})] * 7)
+
+    call_command('cycle_data', 'from:treeherder', sleep_time=0, days=1, chunk_size=3)
+
+    # There should be no jobs after cycling
+    assert Job.objects.count() == 0
+    assert FailureLine.objects.count() == 0
+
+
+def test_cycle_job_model_reference_data(
+    test_repository, failure_classifications, sample_data, sample_push, mock_log_parser
+):
+    job_data = sample_data.job_data[:20]
+    test_utils.do_job_ingestion(test_repository, job_data, sample_push, False)
+
+    # get a list of ids of original reference data
+    original_job_type_ids = JobType.objects.values_list('id', flat=True)
+    original_job_group_ids = JobGroup.objects.values_list('id', flat=True)
+    original_machine_ids = Machine.objects.values_list('id', flat=True)
+
+    # create a bunch of job model data that should be cycled, since they don't
+    # reference any current jobs
+    jg = JobGroup.objects.create(symbol='moo', name='moo')
+    jt = JobType.objects.create(symbol='mu', name='mu')
+    m = Machine.objects.create(name='machine_with_no_job')
+    (jg_id, jt_id, m_id) = (jg.id, jt.id, m.id)
+    call_command('cycle_data', 'from:treeherder', sleep_time=0, days=1, chunk_size=3)
+
+    # assert that reference data that should have been cycled, was cycled
+    assert JobGroup.objects.filter(id=jg_id).count() == 0
+    assert JobType.objects.filter(id=jt_id).count() == 0
+    assert Machine.objects.filter(id=m_id).count() == 0
+
+    # assert that we still have everything that shouldn't have been cycled
+    assert JobType.objects.filter(id__in=original_job_type_ids).count() == len(
+        original_job_type_ids
+    )
+    assert JobGroup.objects.filter(id__in=original_job_group_ids).count() == len(
+        original_job_group_ids
+    )
+    assert Machine.objects.filter(id__in=original_machine_ids).count() == len(original_machine_ids)
+
+
+def test_cycle_job_with_performance_data(
```

Comment:
What do you think about adding a comment here explaining why we need to also test with `performance data` in `test_treeherder_cycling.py`?